plutus-tx/compiler/Language/PlutusTx/Compiler/Names.hs

```diff
@@ -26,8 +27,24 @@ import qualified Data.Text                        as T
 lookupName :: Scope -> GHC.Name -> Maybe PLCVar
 lookupName (Scope ns _) n = Map.lookup n ns
 
+{- |
+Reverses the OccName tidying that GHC does, see 'tidyOccEnv'
+and accompanying Notes.
+
+This is bad, because it makes it much harder to read since the
+disambiguating numbers are gone. However, these appear to be
+non-deterministic (possibly depending on the order in which
+modules are processed?), so we can't rely on them.
+
+Essentially, we just strip off trailing digits.
+This might remove "real" digits added by the user, but
+there's not much we can do about that.
```

Comment:
I'd like to see an explanation of why it is safe for us not to disambiguate things that are distinct.

Reply:
Right. This is only the textual names, which don't matter. We have `Unique`s to actually identify them.